### PR TITLE
use linear interp when pixel ratios don't match

### LINKED ISSUE #1601: Implement a better sprite scaling algorithm
image

We've made it possible to display sprites for non-integer pixel ratios in #1593 but the image scaling isn't very good. 

Maybe related to this [`TODO`](https://github.com/mapbox/mapbox-gl-js/blob/master/js/symbol/sprite_atlas.js#L65-L66)?

ref https://github.com/mapbox/mapbox-gl-js/pull/1593#issuecomment-145964474

### PR DESCRIPTION
**Pull request by ansis**
fix #1601 

- [x] add render test

### CODE REVIEW

File: js/render/draw_symbol.js [hunk 1]
```diff
@@ -125,7 +125,8 @@ function drawSymbol(painter, layer, posMatrix, tile, elementGroups, prefix, sdf)
         texsize = [painter.glyphAtlas.width / 4, painter.glyphAtlas.height / 4];
     } else {
         painter.spriteAtlas.bind(gl, alignedWithMap || painter.options.rotating ||
-            painter.options.zooming || fontScale !== 1 || sdf || painter.transform.pitch);
+            painter.options.zooming || fontScale !== 1 || sdf || painter.transform.pitch ||
+            browser.devicePixelRatio !== painter.spriteAtlas.pixelRatio);
```
Comment (lucaswoj) [inline js/render/draw_symbol.js]: Clean fix! Is there any way we could simplify this expression? Maybe just create some `var` subexpressions?

Comment (ansis) [inline js/render/draw_symbol.js]: What do you have in mind? something like this?

```
var iconIsScaled = fontScale !== 1;
var pixelRatioMismatch = browser.devicePixelRatio !== painter.spriteAtlas.pixelRatio;
painter.spriteAtlas.bind(gl, alignedWithMap || painter.options.rotating ||
            painter.options.zooming || iconIsScaled || sdf || painter.transform.pitch ||
            pixelRatioMismatch);

I'm not sure if that's any simpler

Comment (lucaswoj) [inline js/render/draw_symbol.js]: No, I don't think that's any simpler. I was thinking that there might be some groupings of multiple clauses that would make sense semantically, but if that's not the case :ship:.

Comment (ansis) [inline js/render/draw_symbol.js]: thoughts on this?


        var mapMoving = painter.options.rotating || painter.options.zooming;
        var iconScaled = fontScale != 1 || browser.devicePixelRatio !== painter.spriteAtlas.pixelRatio;
        var iconTransformed = alignedWithMap || painter.transform.pitch;
        painter.spriteAtlas.bind(gl, sdf || mapMoving || iconScaled || iconTransformed);

Comment (lucaswoj) [inline js/render/draw_symbol.js]: Perfect :smile:

**Pull request merged**